src/main/java/cubicchunks/worldgen/dependency/Requirement.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+
+package cubicchunks.worldgen.dependency;
+
+import cubicchunks.util.AddressTools;
+import cubicchunks.worldgen.GeneratorStage;
+
+public class Requirement {
+
+	public int cubeX;
+	public int cubeY;
+	public int cubeZ;
+	public GeneratorStage targetStage;
+	
+	public Requirement(int cubeX, int cubeY, int cubeZ, GeneratorStage targetStage) {
+		this.cubeX = cubeX;
+		this.cubeY = cubeY;
+		this.cubeZ = cubeZ;
+		this.targetStage = targetStage;
+	}
+	
+	public Long getAddress() {
+		return AddressTools.getAddress(cubeX, cubeY, cubeZ);
```

Comment:
Long instead of long is a typo. I'll change it to use CubeCoords in the near future.